stdlib/src/sqlite.rs

```diff
@@ -2726,22 +2726,48 @@ mod _sqlite {
             unsafe { sqlite3_column_name(self.st, pos) }
         }
 
-        fn columns_name(self, vm: &VirtualMachine) -> PyResult<Vec<PyStrRef>> {
+        fn columns_name(self, detect_types: i32, vm: &VirtualMachine) -> PyResult<Vec<PyStrRef>> {
             let count = self.column_count();
             (0..count)
                 .map(|i| {
                     let name = self.column_name(i);
-                    ptr_to_str(name, vm).map(|x| vm.ctx.new_str(x))
+                    let name_str = ptr_to_str(name, vm)?;
+
+                    // If PARSE_COLNAMES is enabled, strip everything after the first '[' (and preceding space)
+                    let processed_name = if detect_types & PARSE_COLNAMES != 0 {
+                        // Find the position of the first '['
+                        if let Some(bracket_pos) = name_str.find('[') {
```

Comment:
```suggestion
                    let processed_name = if detect_types & PARSE_COLNAMES != 0 &&
                        // Find the position of the first '['
                        let Some(bracket_pos) = name_str.find('[') {
```
The latest stable Rust allows to use let chaining